screens/AppNavigator/screens/Dex/DexAddLiquidity.tsx

```diff
@@ -0,0 +1,225 @@
+    props.onChange(amountToSet)
+  }
+
+  return (
+    <View style={tailwind('flex-col w-full h-36 items-center mt-4')}>
+      <View style={tailwind('flex-col w-full h-8 bg-white justify-center')}>
+        <Text style={tailwind('m-4')}>Input</Text>
+      </View>
+      <View style={tailwind('flex-row w-full h-16 bg-white items-center p-4')}>
+        <TextInput
+          style={tailwind('flex-1 mr-4 text-gray-500')}
+          value={props.current.isNaN() ? '' : props.current.toString()}
+          keyboardType='numeric'
+          onChange={event => props.onChange(new BigNumber(event.nativeEvent.text))}
+        />
+        {renderIcon()}
+        <Text style={tailwind('w-12 ml-4 text-gray-500')}>{props.symbol}</Text>
+      </View>
+      <View style={tailwind('w-full bg-white flex-row border-t border-gray-200 h-12 items-center')}>
+        <View style={tailwind('flex flex-row flex-1 ml-4')}>
+          <Text>Balance: </Text>
+          <Text style={tailwind('text-gray-500')}>{props.balance.toString()}</Text>
+        </View>
+        <TouchableOpacity
+          style={tailwind('flex w-12 mr-4')}
+          onPress={onMax}
+        >
+          <Text style={[PrimaryColorStyle.text]}>MAX</Text>
+        </TouchableOpacity>
+      </View>
+    </View>
+  )
+}
+
+function Summary (props: { pair: ExtPoolPairData, sharePercentage: BigNumber }): JSX.Element {
+  const { pair, sharePercentage } = props
+  const RenderRow = (lhs: string, rhs: string): JSX.Element => {
+    return (
+      <View style={tailwind('bg-white p-2 border-b border-gray-200 flex-row items-center w-full h-16')}>
+        <View style={tailwind('flex-1')}>
+          <Text style={tailwind('font-medium')}>{lhs}</Text>
+        </View>
+        <View style={tailwind('flex-1')}>
+          <Text style={tailwind('font-medium text-right text-gray-500')}>{rhs}</Text>
+        </View>
+      </View>
+    )
+  }
+
+  return (
+    <View style={tailwind('flex-col w-full items-center mt-4')}>
+      <View style={tailwind('bg-white p-2 border-b border-gray-200 flex-row items-center w-full h-16')}>
+        <View style={tailwind('flex-1')}>
+          <Text style={tailwind('font-medium')}>Price</Text>
+        </View>
+        <View style={tailwind('flex-1 flex-col')}>
+          <Text style={tailwind('font-medium text-right text-gray-600')}>{pair.aToBRate.toString()} {pair.aSymbol} per {pair.bSymbol}</Text>
+          <Text style={tailwind('font-medium text-right text-gray-600')}>{pair.bToARate.toString()} {pair.bSymbol} per {pair.aSymbol}</Text>
+        </View>
+      </View>
+      {RenderRow('Share of pool', sharePercentage.times(100).toString())}
+      {RenderRow(`Pooled ${pair.aSymbol}`, pair.tokenA.reserve)}
+      {RenderRow(`Pooled ${pair.bSymbol}`, pair.tokenB.reserve)}
+    </View>
+  )
+}
+
+function ContinueButton (props: { enabled: boolean, onPress: () => void }): JSX.Element {
+  const buttonColor = props.enabled ? PrimaryColorStyle.bg : { backgroundColor: 'gray' }
```

Comment:
You can use this component for PrimaryButton with validation -https://github.com/DeFiCh/wallet/blob/e29908f0f284a18aa9b2ca99fb8555a138d51203/components/PrimaryButton.tsx